notifications_utils/formatters.py

```diff
@@ -47,9 +46,15 @@
     ('&rpar;', "▶️🐦🥴"),
 )
 
-# The Mistune URL regex only matches URLs at the start of a string,
-# using `^`, so we slice that off and recompile
-url = re.compile(mistune.InlineGrammar.url.pattern[1:])
+url = re.compile(
+    r'(?i)'  # case insensitive
+    r'(https?:\/\/)?'  # optional http:// or https://
+    r'([\w\-])+\.{1}'  # one or more (sub)domains
+    r'([a-z]{2,63})'  # top-level domain
+    r'([\/\w-]*)*\/?'  # path with optional trailing slash
+    r'\??([^#\n\r\<]*)?'  # optional query string
+    r'#?([^\n\r\<]*)'  # optional #fragment
+)
```

Comment:
I don't think this code will handle terminating `.,:` etc too

Reply:
Note that we decided to scope this to not changing the regex used for emails.

That would make this change even more complicated because then we’d have to make sure not to break any existing emails.